### PR TITLE
35: Move orchestrator db schema here and rewire integration test to u…

### PR DESCRIPTION
**Pull request by lfse-slafleur**
…se computation engine as a submodule.

Needs to be coordinated with https://github.com/Project-OMOTES/computation-engine/pull/43

### CODE REVIEW

File: integration_test/setup.sh [hunk 1]
```diff
@@ -1,33 +1,22 @@
 #!/bin/bash
 
-. .env
+CURRENT_WORKDIR=$PWD
+COMPUTATION_ENGINE="../computation-engine-at-orchestrator"
+ENV_FILE="${CURRENT_WORKDIR}/.env"
```
Comment (MarkTNO) [inline integration_test/setup.sh:5]: Another name like `.env.tests` would avoid replacing the existing `.env` file when running locally.

File: unit_test/test_main.py [hunk 2]
```diff
@@ -148,6 +161,220 @@ def test__cleanup_barrier__cleaning_barrier(self) -> None:
...
+class OrchestratorTest(unittest.TestCase):
+    def test__new_job_submitted_handler__fully_new_job(self) -> None:
+        # Arrange
+        omotes_orchestrator_if = Mock()
+        jobs_broker_if = Mock()
+        celery_if = Mock()
+        celery_if.start_workflow.return_value = "celery_id"
+        postgresql_if = Mock()
+        postgresql_if.job_exists.return_value = False
+        workflow_manager = Mock()
```
Comment (MarkTNO) [inline unit_test/test_main.py:173]: duplicate code: lines 173-207 and 233-367, maybe put in function?

**Review by MarkTNO — COMMENTED** [review]
Again really nice to have these tests available, extremely useful!

A couple of general remarks:
- There are a lot of console prints during the integration test, this would probably make it harder to track down issues. Is it possible to reduce the prints?
- Should we divide `unit_test/test_main.py` into several files?
- I really appreciate your effort in creating 'win32' cmd files, however I don't use them.. Since two weeks I run everything under WSL (file system and venv) and before I was using 'git bash' for the shell scripts. So as far as I am concerned we can leave out these cmd files. It is much better to develop with the same scripts as will be used on ci.

File: integration_test/setup.sh (hunk 1 above)
Comment (lfse-slafleur) [reply integration_test/setup.sh]: Fixed in d0de2316f6858f0cde975d939ec42b49cdfcf206

File: unit_test/test_main.py (hunk 2 above)
Comment (lfse-slafleur) [reply unit_test/test_main.py]: I found this one difficult. I have reduced it some here: https://github.com/Project-OMOTES/orchestrator/pull/36/commits/f7df22d7b74a85866100012c4f989ba4c704279e

However, most values are required throughout the function so I am afraid that further deduplication will hurt readability of the test. What do you think? Is it reduced enough like this?

**Comment by lfse-slafleur** [comment]
>     * There are a lot of console prints during the integration test, this would probably make it harder to track down issues. Is it possible to reduce the prints?
Quieted down by setting the log level.


>     * Should we divide `unit_test/test_main.py` into several files?
I would vote not yet. This PR is quite big already. Also, a large test file is usually a sign the 'file under test' (`main.py`) is too big and should be split up. In other words, if we split up `main.py`, the unit test file will be reduced as well. But I would vote to do this later.

>     * I really appreciate your effort in creating 'win32' cmd files, however I don't use them.. Since two weeks I run everything under WSL (file system and venv) and before I was using 'git bash' for the shell scripts. So as far as I am concerned we can leave out these cmd files. It is much better to develop with the same scripts as will be used on ci.
Michiel requires them.

**Comment by lfse-slafleur** [comment]
Also, there remains an issue that sometimes causes the integration test to fail. It is an issue within aio-pika. I have captured the issue here https://github.com/mosquito/aio-pika/issues/630 and will solve it myself. Unsure when it gets solved and merged in. Should not be an issue on production systems as the main reason it fails in the integration test is because the jobs finish so extremely fast. Jobs within omotes takes at least a couple of seconds at the moment. Should be fixed ASAP (high priority). Reminder captured here: https://github.com/Project-OMOTES/orchestrator/issues/37

File: unit_test/test_main.py (hunk 2 above)
Comment (MarkTNO) [reply unit_test/test_main.py]: yes, much better

**Comment by MarkTNO** [comment]
Thanks for resolving!  
One more question about the win cmd files. Would Michiel be able to work with the bash files in git bash or powershell?

If not, should we provide cmd equivalents for all scripts? For instance the `integration_test/ci` scripts? And all scripts in the computation engine?

**Review by MarkTNO — APPROVED** [review]

**Comment by lfse-slafleur** [comment]
> Thanks for resolving! One more question about the win cmd files. Would Michiel be able to work with the bash files in git bash or powershell?
> 
> If not, should we provide cmd equivalents for all scripts? For instance the `integration_test/ci` scripts? And all scripts in the computation engine?

I am not sure exactly what his current working process is but I know he can work with cmd and ps files.
Eventually when we uses the other scripts, I propose we port them then. For now, I ported the 'easy' ones already.

**Pull request merged**